pandas/core/indexes/base.py

```diff
@@ -2190,19 +2190,24 @@ def __or__(self, other):
     def __xor__(self, other):
         return self.symmetric_difference(other)
 
-    def _get_consensus_name(self, other):
+    def _get_setop_name(self, other):
+        # GH 9943 9862
         """
-        Given 2 indexes, give a consensus name meaning
-        we take the not None one, or None if the names differ.
-        Return a new object if we are resetting the name
+        Given 2 indexes, give the name appropriate for set
+        operations meaning we take the name if both names
+        are the same, or None if the names differ.
         """
-        if self.name != other.name:
-            if self.name is None or other.name is None:
-                name = self.name or other.name
-            else:
-                name = None
-            if self.name != name:
-                return self._shallow_copy(name=name)
+        return self.name if self.name == other.name else None
```

Comment:
this logic already exists in ``pandas.core.ops.get_op_result_name`` I would just use this as a function and not as a method

Reply:
A function added just a few days ago!  Will make that change.